wondrous-app/components/CreateEntity/CreateEntityModal/TaskTemplatePicker/index.tsx

```diff
@@ -0,0 +1,343 @@
+                      </TaskTemplateRewardBox>
+                    ) : null}
+                  </TaskTemplateRewardContainer>
+                </TaskTemplateListItems>
+              </TaskTemplateContainer>
+            );
+          })
+        ) : (
+          <TaskTemplateLabelValue
+            style={{
+              fontSize: '15px',
+            }}
+          >
+            No templates yet!
+          </TaskTemplateLabelValue>
+        )}
+      </TaskTemplateActionPopper>
+    );
+  };
+
+  const renderContainer = () => {
+    switch (saveOrOpen) {
+      case 'open':
+        return <TaskTemplateOpen />;
+      default:
+        return (
+          <TaskTemplateDefaultPopper
+            open={open}
+            anchorEl={anchorEl}
+            placement="top-start"
+            disablePortal={true}
+            style={sizingForPopperContainer()}
+          >
+            <TaskTemplateOptionsLabel
+              onClick={() => {
+                setSaveOrOpen('save');
+              }}
+            >
+              Save new template
+            </TaskTemplateOptionsLabel>
+            <TaskTemplateOptionsLabel
+              onClick={() => {
+                setSaveOrOpen('open');
+              }}
+            >
+              Open existing template
+            </TaskTemplateOptionsLabel>
+          </TaskTemplateDefaultPopper>
+        );
+    }
+  };
+
+  const sizingForPopperContainer = () => {
+    switch (saveOrOpen) {
+      case 'save':
+        return { width: 500, padding: '0px' };
+      case 'open':
+        return { height: 600, width: 500, padding: '24px' };
+      default:
+        return null;
+    }
+  };
+
+  const getPaymentMethodData = (id) => paymentMethods.find((payment) => payment.id === id);
+
+  const labels = ['title', 'description', 'reward', 'assignee'];
+  return (
+    <TaskTemplateClickAway onClickAway={handleClickAway}>
+      <div>
+        <TaskTemplateModal open={open} disabled={!options || disabled} onClick={handleClick}>
+          <TaskTemplateLabelWrapper>
+            {usedTemplate ? <TaskTemplateIcon color={'474747'} /> : null}
+            <TaskTemplateLabel>{usedTemplate?.name ?? `Select Template`}</TaskTemplateLabel>
+          </TaskTemplateLabelWrapper>
+          {usedTemplate ? (
+            <TaskTemplateDeleteIcon
+              onClick={(e) => {
+                e.preventDefault();
+                e.stopPropagation();
+                setUsedTemplate(null);
+              }}
+            />
+          ) : (
+            <TaskTemplateArrowIcon />
+          )}
+        </TaskTemplateModal>
+
+        {saveOrOpen === 'save' ? (
+          <TaskTemplateDefaultPopper
+            open={open}
+            anchorEl={anchorEl}
+            placement="top-start"
+            disablePortal={true}
+            style={sizingForPopperContainer()}
+          >
+            <div style={{ padding: '24px' }}>
```

Comment:
applies for the whole component: can we reduce the inline styles and move them to styles.tsx ?